src/ray/gcs/gcs_server/gcs_actor_scheduler.h

```diff
@@ -0,0 +1,261 @@
+  /// It provides interface called `CreateActor` to push a normal task to the remote
+  /// worker to create the real actor. Through this class, we can easily get the WorkerID,
+  /// Endpoint, NodeID and the associated actor of the remote worker.
+  class GcsLeasedWorker : public std::enable_shared_from_this<GcsLeasedWorker> {
+   public:
+    /// Create a GcsLeasedWorker
+    ///
+    /// \param address the Address of the remote leased worker.
+    /// \param resources the resources that leased from the remote node(raylet).
+    /// \param client_call_manager is shared by all `CoreWorkerClient`s.
+    explicit GcsLeasedWorker(
+        rpc::Address address,
+        const google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry> &resources,
+        boost::asio::io_context &io_context, rpc::ClientCallManager &client_call_manager)
+        : address_(std::move(address)),
+          resources_(resources),
+          io_context_(io_context),
+          client_call_manager_(client_call_manager) {}
+    virtual ~GcsLeasedWorker() = default;
+
+    /// Create the specified actor on the leased worker asynchronously.
+    ///
+    /// \param actor will be created on the remote leased worker.
+    /// \param on_done Callback of the create request, which will be invoked only when
+    /// success, otherwise it will never be triggered.
+    virtual void CreateActor(std::shared_ptr<GcsActor> actor,
+                             const std::function<void()> &on_done);
+
+    /// Get the Address of this leased worker.
+    const rpc::Address &GetAddress() const { return address_; }
+
+    /// Get the ip address of this leased worker.
+    const std::string &GetIpAddress() const { return address_.ip_address(); }
+
+    /// Get the listening port of the leased worker at remote side.
+    uint16_t GetPort() const { return address_.port(); }
+
+    /// Get the WorkerID of this leased worker.
+    WorkerID GetWorkerID() const { return WorkerID::FromBinary(address_.worker_id()); }
+
+    /// Get the NodeID of this leased worker.
+    ClientID GetNodeID() const { return ClientID::FromBinary(address_.raylet_id()); }
+
+    /// Get the actor which is creating on this leased worker.
+    std::shared_ptr<GcsActor> GetActor() const { return actor_; }
+
+   private:
+    /// Get or create CoreWorkerClient to communicate with the remote leased worker.
+    /// If the client_ is nullptr then create and cache it, otherwise return the client_.
+    std::shared_ptr<rpc::CoreWorkerClient> GetOrCreateClient() {
+      if (client_ == nullptr) {
+        rpc::Address address;
+        address.set_ip_address(GetIpAddress());
+        address.set_port(GetPort());
+        client_ = std::make_shared<rpc::CoreWorkerClient>(address, client_call_manager_);
+      }
+      return client_;
+    }
+
+    /// Reset the client so that we can create a new one when retry creating actors.
+    void ResetClient() { client_.reset(); }
```

Comment:
In this case the ip_address and port would change since it will be talking with a different worker? In that case should we create a new GcsLeasedWorker instead, which seems cleaner?